src/util/kaldi-table.cc

```diff
@@ -22,6 +22,21 @@
 
 namespace kaldi {
 
+bool ExtractRangeSpecifier(const std::string &line,
+                           std::string *data_rxfilename,
+                           std::string *range) {
+  if (line.empty() || line[line.size()-1] != ']')
+    KALDI_ERR << "ExtractRangeRspecifier called wrongly.";
+  std::vector<std::string> splits;
+  SplitStringToVector(line, "[", false, &splits);
+  if (splits.size() == 2 && !splits[0].empty() && !splits[1].empty()) {
```

Comment:
`splits[1].empty()` is tautologically `true` if evaluated, as the last character known to be `]` will be included into last split.
The condition as written allows for empty range spec `[]` -- is it legit?